settingshelper.py

```diff
@@ -283,3 +284,23 @@ def get_git_commit(base_dir):
         return out.strip().decode('ascii')
     except OSError:
         pass
+
+
+def update_redis_location_for_tests(settings_caches):
+    if not is_testing():
+        raise Exception("Attempt to update Redis settings outside of tests")
+
+    for name, config in settings_caches.items():
+        if not config.get("BACKEND", "").startswith("django_redis"):
+            continue
+
+        test_location = config.get("TEST_LOCATION")
+        if not test_location:
+            logging.warning(
+                "Unable to set Redis DB in '%(name)s' cache for tests. Using '%(location)s'.\n"
+                "\tTo configure a separate Redis DB for tests add a 'TEST_LOCATION' to the"
+                " '%(name)s' cache configuration.", {"name": name, "location": config["LOCATION"]}
+            )
+
+        logging.info("Using '%s' connection for Redis", test_location)
+        config["LOCATION"] = test_location
```

Comment:
I personally would prefer this raised an error rather than logging warning, and I'd prefer it to not log anything on the happy path.

Also, I think there's a bug (it will overwrite `config["LOCATION"]` even if `not test_location` is true).
```suggestion
        if test_location:
            config["LOCATION"] = test_location
        else:
            logging.warning(
                "Unable to set Redis DB in '%(name)s' cache for tests. Using '%(location)s'.\n"
                "\tTo configure a separate Redis DB for tests add a 'TEST_LOCATION' to the"
                " '%(name)s' cache configuration.", {"name": name, "location": config["LOCATION"]}
            )
```

Reply:
thanks that was the intention